components/eamxx/src/diagnostics/atm_backtend.cpp

```diff
@@ -0,0 +1,91 @@
+#include "diagnostics/atm_backtend.hpp"
+
+#include <ekat/kokkos/ekat_kokkos_utils.hpp>
+
+#include "share/util/scream_universal_constants.hpp"
+
+namespace scream {
+
+AtmBackTendDiag::AtmBackTendDiag(const ekat::Comm &comm,
+                                 const ekat::ParameterList &params)
+    : AtmosphereDiagnostic(comm, params) {
+  EKAT_REQUIRE_MSG(params.isParameter("Tendency Name"),
+                   "Error! AtmBackTendDiag requires 'Tendency Name' in its "
+                   "input parameters.\n");
+
+  m_name = m_params.get<std::string>("Tendency Name");
+}
+
+std::string AtmBackTendDiag::name() const { return m_name + "_atm_tend"; }
+
+void AtmBackTendDiag::set_grids(
+    const std::shared_ptr<const GridsManager> grids_manager) {
+  using namespace ekat::units;
+
+  const auto &gname = m_params.get<std::string>("grid_name");
+  add_field<Required>(m_name, gname);
+}
+
+void AtmBackTendDiag::initialize_impl(const RunType /*run_type*/) {
+  const auto &f   = get_field_in(m_name);
+  const auto &fid = f.get_header().get_identifier();
+  const auto &gn  = fid.get_grid_name();
+
+  // Sanity checks
+  using namespace ShortFieldTagsNames;
+  const auto &layout = fid.get_layout();
+  EKAT_REQUIRE_MSG(
+      f.data_type() == DataType::RealType,
+      "Error! AtmBackTendDiag only supports Real data type field.\n"
+      " - field name: " +
+          fid.name() +
+          "\n"
+          " - field data type: " +
+          e2str(f.data_type()) + "\n");
+
+  using namespace ekat::units;
+  // The units are the same except per second
+  auto diag_units = fid.get_units() / s;
+  // TODO: set the units string correctly by appending "/s"
```

Comment:
That's done automatically now. It may be looking a bit cumbersome if the input field units are not a single simbol. E.g., if input field is in `W/m2`, your diag will have units `(W/m2)/s`. If you want units to always be `<input_units> / s`, without parentheses, then you need to do
```
Units diag_units (fid.get_units()/s, fid.get_units().to_string() + " / s");
```